apps/sms/views/shared/js/settings.js

```diff
@@ -67,9 +70,37 @@ var Settings = {
       }
     }
 
+    //create lock for SEND_READ_REPORT_KEY
+    keyHandlerSet[this.SEND_READ_REPORT_KEY] = this.toSendReadReport.bind(this);
+
     for (var key in keyHandlerSet) {
       setHandlerMap(key);
     }
+
+    //add Observer for SEND_READ_REPORT_KEY
+    settings.addObserver(this.SEND_READ_REPORT_KEY, function (event) {
+      Settings.sendReadReportStatus = event.settingValue;
+      console.log(Settings.sendReadReportStatus);
+    });
+  },
+
+  sendReadReport: function toSendReadReport() {
+    var deferred = Utils.Promise.defer();
+    if(Settings.sendReadReportStatus != null) {
+      deferred.resolve(Settings.sendReadReportStatus);
+    } else {
+      deferred.reject('ERROR: Settings.sendReadReportStatus unavailable');
+    };
+
+    return deferred.promise;
+  },
+
+  //Set sendReadReportStatus for SEND_READ_REPORT_KEY
+  toSendReadReport: function toSendReadReport(status) {
+    if (!isNaN(status)) {
+      Settings.sendReadReportStatus = status;
+      console.log(Settings.sendReadReportStatus);
+    }
   },
```

Comment:
Regarding the chaining of api calls, in which situation, when we should go for it? 
When we are accessing IndexDB or some complex situation , we should go for chaining and at other place it's fine without it?